hack/test-go.sh

```diff
@@ -121,7 +121,7 @@ done
 shift $((OPTIND - 1))
 
 # Use eval to preserve embedded quoted strings.
-eval "goflags=(${KUBE_GOFLAGS:-})"
```

Comment:
I think it's debateable whether the integration tests should set the test build tag... I see that mesos uses the `unit_test` tag, which may be better in this case... WDYT?